app/src/test/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntryEffectHandlerTest.kt

```diff
@@ -334,10 +333,56 @@ class BloodSugarEntryEffectHandlerTest {
     )
     val updateBloodSugarEntry = UpdateBloodSugarEntry(
         bloodSugarMeasurementUuid = bloodSugarMeasurementUuid,
-        bloodSugarReading = 145,
-        measurementType = bloodSugar.reading.type,
         userEnteredDate = updatedDate,
-        prefilledDate = updatedDate
+        prefilledDate = updatedDate,
+        bloodSugarReading = updateBloodSugarReading
+    )
+
+    whenever(userSession.loggedInUserImmediate()).doReturn(user)
+    whenever(facilityRepository.currentFacilityImmediate(user)).doReturn(facility)
+    whenever(bloodSugarRepository.measurement(bloodSugarMeasurementUuid)).doReturn(bloodSugar)
+
+    // when
+    testCase.dispatch(updateBloodSugarEntry)
+
+    // then
+    testCase.assertOutgoingEvents(BloodSugarSaved(updateBloodSugarEntry.wasDateChanged))
+    verifyZeroInteractions(ui)
+  }
+
+  @Test
+  fun `update hba1c blood sugar entry when update blood sugar entry effect is received`() {
+    // given
+    val user = PatientMocker.loggedInUser(uuid = UUID.fromString("50da2a45-3680-41e8-b46d-8a5896eadce0"))
+    val facility = PatientMocker.facility(uuid = UUID.fromString("7fabe36b-8fc3-457d-b9a8-68df71def7bd"))
+    val patientUuid = UUID.fromString("260a831f-bc31-4341-bc0d-46325e85e32d")
+
+    val date = LocalDate.parse("2020-02-14")
+    val updatedDate = LocalDate.parse("2020-02-12")
+
+    val bloodSugarReading = BloodSugarReading(5.6.toFloat(), HbA1c)
+    val updateBloodSugarReading = bloodSugarReading.copy(value = 6.2.toFloat())
```

Comment:
Instead of converting values using `toFloat()`, can we not using floating values directly?